gpu_direct_rdma_access.c

```diff
@@ -146,6 +149,11 @@ struct rdma_buffer {
     struct rdma_device *rdma_dev;
 };
 
+static inline
+int is_server(struct rdma_device *device)
+{
+	return device->srq != NULL;
```

Comment:
is_server() should return TRUE (1) if it is a server indeed.
the check here looks the other way around.... and later you use it as if it was "if (is_client() ..)"